test/conftest.py

```diff
@@ -0,0 +1,22 @@
+import random
+
+import pytest
+import torch
+
+
+@pytest.fixture(autouse=True)
+def prevent_leaking_rng():
+    # Prevent each test from leaking the rng to all other test when they call
+    # torch.manual_seed() or random.seed().
```

Comment:
This is taken from torchvision

Reply:
My understanding of what this is doing, and why it works:

1. Because we have `autouse=True`, this fixture is used for _all_ tests, including our decoder tests.
2. As this is autoused, we execute lines 8--15 before all tests. That effectively saves the current RNG state for torch, random and torch.cuda.
3. We yield on line 17, which lets the actual test execute. That test may itself change the state of the above.
4. After the test runs, we return back on line 19, and restore the state we captured above.

Is my understanding correct?

Reply:
Yup, that's it. This is a use of pytest' fixtures as a scoped setup/teardown mechanism.